src/hotspot/share/runtime/java.cpp

```diff
@@ -96,9 +96,10 @@
 GrowableArray<Method*>* collected_profiled_methods;
 
 int compare_methods(Method** a, Method** b) {
-  // %%% there can be 32-bit overflow here
-  return ((*b)->invocation_count() + (*b)->compiled_invocation_count())
-       - ((*a)->invocation_count() + (*a)->compiled_invocation_count());
+  // invocation_count() may have overflowed already. Interpret it's result as
+  // unsigned int to shift the limit of meaningless results by a factor of 2.
```

Comment:
Code is fine, but this comment doesn't make sense to me. The result is the same with your version. But it has the advantage that it avoids signed integer overflow (undefined behavior).

Reply:
I agree. The comments could be misleading. They are gone.